modules/new_profile/php/new_profile.class.inc

```diff
@@ -170,21 +161,17 @@ class New_Profile extends \NDB_Form
         $maxYear   = (isset($endYear, $ageMin)) ? $endYear - $ageMin : null;
 
         // Get sites for the select dropdown
-        $user_list_of_sites = $user->getData('CenterIDs');
-        $num_sites          = count($user_list_of_sites);
-        $psc_labelOptions   = array();
-        if ($num_sites > 1) {
-            foreach ($user_list_of_sites as $key => $siteID) {
-                $center = $DB->pselectRow(
-                    "SELECT CenterID as ID, Name FROM psc WHERE CenterID =:cid",
-                    array('cid' => $siteID)
-                );
-                if (!is_null($center)) {
-                    $psc_labelOptions[$siteID] = $center['Name'];
-                }
+        $userSites = $user->getData('CenterIDs');
+        $sites     = array();
+        foreach ($userSites as $key => $siteID) {
```

Comment:
instead of multiple call to the DB, you could do a query with `WHERE CenterID IN (...)` and provide a comma separated list of values. $userSites is obtained from the code which get it from the DB.